__tests__/app.test.js

```diff
@@ -163,14 +194,72 @@ describe("/api/articles", () => {
             });
         });
     })
-    it("GET 200: returned array should be sorted by date in descending order", () => {
+    it("GET 200: returned array should be sorted by date in descending order by default", () => {
         return request(app)
         .get("/api/articles")
         .expect(200)
         .then(({ body }) => {
             expect(body.articles).toBeSortedBy("created_at", { descending: true, })
         });
     });
+    it("GET 200: takes a sort_by query and responds with articles sorted by the given column name in the default order", () => {
+        return request(app)
+        .get("/api/articles?sort_by=author")
+        .expect(200)
+        .then(({ body }) => {
+            expect(body.articles).toBeSortedBy("author", { descending: true })
+        });
+    });
+    
+    it("GET 200: responds with articles sorted in ascending or descending order by the default column", () => {
+        return request(app)
+        .get("/api/articles?order=asc")
+        .expect(200)
+        .then(({ body }) => {
+            expect(body.articles).toBeSortedBy("created_at", { descending: false })
+        });
+    });
+    it("GET 200: responds with articles sorted in ascending or descending order by a chosen valid column", () => {
+        return request(app)
+        .get("/api/articles?sort_by=topic&order=asc")
+        .expect(200)
+        .then(({ body }) => {
+            expect(body.articles).toBeSortedBy("topic", { descending: false })
+        });
+    });
+    it("GET 200: responds with articles sorted in ascending or descending order sorted by a combined total", () => {
+        return request(app)
+        .get("/api/articles?sort_by=comment_count&order=desc")
+        .expect(200)
+        .then(({ body }) => {
+            expect(body.articles).toHaveLength(13)
+            expect(body.articles).toBeSortedBy("comment_count", { descending: true })
+        });
+    });
+    it("GET 400: responds with an error message when given an invalid sort_by query", () => {
+        return request(app)
+        .get("/api/articles?sort_by=cats")
+        .expect(400)
+        .then(({ body }) => {
+            expect(body.msg).toBe("Bad Request");
+        });
+    });
+    it("GET 400: responds with an error message when given an invalid order parameter", () => {
+        return request(app)
+        .get("/api/articles?order=upwards")
+        .expect(400)
+        .then(({ body }) => {
+            expect(body.msg).toBe("Bad Request");
+        });
+    });
+    it("GET 400: responds with an error message when invalid queries are included alongside valid ones", () => {
+        return request(app)
+        .get("/api/articles?sort_by=topic&hello")
+        .expect(400)
+        .then(({ body }) => {
+            expect(body.msg).toBe("Bad Request");
+        });
+    });
 })
```

Comment:
Nice Katherine!! I think there is only one test missing - what if I want to use the `sort_by` and `order` queries together?